Sources/Persist/Persister.swift

```diff
@@ -155,6 +156,7 @@ public final class Persister<Value> {
         valueRemover: @escaping ValueRemover,
         defaultValue: @autoclosure @escaping () -> Value,
         defaultValuePersistBehaviour: DefaultValuePersistOption = [],
+        logger: PersistLogger? = nil,
```

Comment:
Now now I see this haha.. But personally this is another great reason I don't like it in init/function calls unless something will 'happen' during init, otherwise it's almost never required.

Reply:
Is the alternative to have it as a public property and set it after init? I don't think I like the idea of it being part of the public API post-init.

Reply:
yeah that's how I personally feel it _should_ be but its subjective 100%. I think for me, this suggests the logger is a PART of the API.

Say in the future you want to change this API, you have to consider this each time.

Moving it to post-init (love that term btw haha) means you don't have this issue ever. Also if you had multiple init's you have to include this in ALL of them.

I guess for me, the logging is completely secondary.

Reply:
I was thinking I would add a static var somewhere to allow for the default log to be set, so this hopefully wouldn't need to be used much.

As you can probably see I haven't worked on this very much lately because I've not felt a need for a it. I was planning to revisit it once I'd done some more exploration of the code generation because – as you say – this is then needed in every init.